Activate the linters for the entire code base

kevgo opened the pull request:
When we renamed the command to `gt` we lost almost the entire linter coverage.

@charlierudolph

charlierudolph commented:
Do we also need to list the `lib` folder?

kevgo commented:
The `lib` folder is somehow automatically included when we lint the `cmd` folder.

charlierudolph commented:
Then can we lint just the entry point (`cmd/gt`) as everything else will be automatically included?

kevgo commented:
Strangely enough, running the linters on `cmd` did make them run for a long time, but they didn't seem to lint the `lib` folder at all. So I have added a call to lint `lib` directly, and fixed the thousands of linter errors in the previous PR. At least we now have a working linter!

charlierudolph commented on bin/lint_go:
```diff
@@ -6,4 +6,9 @@ if [ -n "$(gofmt -d cmd lib 2>&1)" ]; then
+# - gocyclo: will be enabled and the code refactored later
+# - goconst: let's enable this after more generic refactorings
+gometalinter.v1 -D gas -D gocyclo -D goconst lib/...
+gometalinter.v1 cmd --deadline=20s
```
Is the deadline still needed here? We don't have that many files in cmd?

kevgo replied:
Good catch, I have removed it. This still runs on my Macbook. This is the weakest machine in our toolkit, since it has a mobile processor.
